.github/workflows/docker-image-build-push.yml

```diff
@@ -0,0 +1,40 @@
+name: Build and publish Docker image
+
+on:
+  push:
+    branches: [ main ]
```

Comment:
we mainly work on `develop` right now, should we have a separate docker image for develop as well?